src/aks-preview/azext_aks_preview/tests/latest/test_localdns_profile.py

```diff
@@ -0,0 +1,101 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import unittest
+from unittest.mock import Mock
+from azext_aks_preview.agentpool_decorator import AKSPreviewAgentPoolContext, AKSPreviewAgentPoolModels, AKSPreviewAgentPoolUpdateDecorator
+from azure.cli.command_modules.acs.agentpool_decorator import AKSAgentPoolParamDict
+from azure.cli.command_modules.acs._consts import AgentPoolDecoratorMode, DecoratorMode
+from azure.cli.core.azclierror import InvalidArgumentValueError, MutuallyExclusiveArgumentError
+from types import SimpleNamespace
+import tempfile
+import json
+
+class TestLocalDNSProfile(unittest.TestCase):
```

Comment:
Great to see you've added a thorough unit test for this. Please also include a scenario test, see examples in test_aks_command.py.